### PR TITLE
solution

### PR DESCRIPTION
**Pull request by olenitut**

### CODE REVIEW

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -7,7 +7,23 @@
...
+  actions.forEach(action => {
+    if (action.type === 'addProperties') {
+      Object.assign(currentState, action.extraData);
+    } else if (action.type === 'clear') {
+      currentState = {};
+    } else {
+      action.keysToRemove.forEach(key => {
+        delete currentState[key];
+      });
+    }
```
Comment (Bohdan-Maksymiuk) [inline src/transformStateWithClones.js:22]: avoid such nesting `else if` conditions
This code is hard to read and maintain

**Review by Bohdan-Maksymiuk — CHANGES_REQUESTED** [review]
Good job 👍 
Left the same comment as below previous task
fix it and I'll approve

File: src/transformStateWithClones.js [hunk 2]
```diff
@@ -7,7 +7,27 @@
...
+      case 'addProperties':
+        Object.assign(currentState, action.extraData);
+        break;
+      case 'clear':
+        currentState = {};
+        break;
+      default:
+        action.keysToRemove.forEach(key => {
+          delete currentState[key];
+        });
```
Comment (Bohdan-Maksymiuk) [inline src/transformStateWithClones.js:24]: it should be under `clear` case

**Review by Bohdan-Maksymiuk — CHANGES_REQUESTED** [review]

**Review by Viktor-Kost — APPROVED** [review]
Nice!

File: src/transformStateWithClones.js [hunk 3]
```diff
@@ -7,7 +7,27 @@
...
+      case 'addProperties':
+        Object.assign(currentState, action.extraData);
+        break;
+      case 'clear':
+        currentState = {};
+        break;
+      case 'removeProperties':
+        action.keysToRemove.forEach(key => {
+          delete currentState[key];
+        });
```
Comment (Viktor-Kost) [inline src/transformStateWithClones.js:24]: Similarly to previous tasks, please, add a `default `case

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]
You are so close! 
Just make a little update

**Review by Viktor-Kost — APPROVED** [review]
Nice!

**Review by Virass — APPROVED** [review]

**Pull request closed without merging**